js/src/Ice/ObjectAdapter.js

```diff
@@ -492,27 +467,17 @@ export class ObjectAdapter {
         }
     }
 
-    computePublishedEndpoints() {
-        let p;
+    async computePublishedEndpoints() {
+        let endpoints = [];
         if (this._routerInfo !== null) {
-            p = this._routerInfo.getServerEndpoints().then(endpts => {
-                //
-                // Remove duplicate endpoints, so we have a list of unique endpoints.
-                //
-                const endpoints = [];
-                endpts.forEach(endpoint => {
-                    if (endpoints.findIndex(value => endpoint.equals(value)) === -1) {
-                        endpoints.push(endpoint);
-                    }
-                });
-                return endpoints;
-            });
+            endpoints = await this._routerInfo.getServerEndpoints();
```

Comment:
The only reason createObjectAdapter is async is because sometimes we call getServerEndpoints on the router?

Reply:
Yes, it is only for routed adapters.